library/Vanilla/Embeds/GiphyEmbed.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0
+ */
+
+namespace Vanilla\Embeds;
+
+use Exception;
+
+/**
+ * giphy Embed.
+ */
+class GiphyEmbed extends Embed {
+
+    /** @inheritdoc */
+    protected $domains = ['giphy.com', 'media.giphy.com', 'gph.is'];
+
+    /**
+     * giphyEmbed constructor.
+     */
+    public function __construct() {
+        parent::__construct('giphy', 'gif');
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function matchUrl(string $url) {
+        $data = [];
+        $domain = parse_url($url, PHP_URL_HOST);
+
+        if ($this->isNetworkEnabled()) {
+
+            $post = $this->parseURL($url);
+
+            $oembedData = $this->oembed("https://giphy.com/services/oembed?url=".urlencode($url));
+            if ($oembedData) {
+                $data = $this->normalizeOembed($oembedData);
+            }
+
+            // Obtain the real post ID from media url provided by oembed call
+            if ($domain == 'gph.is') {
+                $post = $this->parseURL($data['url']);
+            }
+
+             if ($post) {
+                $data['attributes']['postID'] = $post['postID'];
+            }
+        }
+
+        return $data;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function renderData(array $data): string {
+        $height = $data['height'] ?? 1;
+        $encodedHeight = htmlspecialchars($height);
+
+        $width = $data['width'] ?? 1;
+        $encodedWidth = htmlspecialchars($width);
+
+        $padding = ($encodedHeight/$encodedWidth) * 100;
```

Comment:
Before attempting to divide the height by the width, you should verify they're numerical values (`is_numeric`). Otherwise, I think you'll possibly be attempting to divide using actual strings.

Reply:
If width or height is not numeric or not provided, i'm not sure what a proper fallback would be.  Stephane suggested rendering it as an image, but I believe that's out of this scope for this issue